client/client_test.go

```diff
@@ -199,7 +199,7 @@ var readResponseTests = []struct {
 	// excessively long chunk.
 	{"HTTP/1.1 200 Ok\r\nTransfer-encoding: chunked\r\n\r\n" +
 		"004\r\n1234\r\n" +
-		"1000000000000000000001\r\n@\r\n" +
+		"1000000000000001\r\n@\r\n" +
```

Comment:
I think this needs to be restored to the original error of "http chunk too large" and then we need to troubleshoot that error as I think shortening that data is causing a red herring error.

Reply:
I think we can fix this if we refer to https://en.wikipedia.org/wiki/Chunked_transfer_encoding about chunked transfer encoding for http, i think we need to change the red of the data that is provided there, not just the `1000000000000000000001` since the other data specifies how long the chunk is.

Reply:
From what I can tell, that part of the code did not exist until go1.6, and this code, according to the .travis.yml file, was only being tested against go1.5 and earlier (https://cs.opensource.google/go/go/+/release-branch.go1.5:src/net/http/internal/chunked.go). I think we should just comment out that particular test data with a comment that references the above about it not existing and linking to the code etc.

Reply:
@coreydaley I agree that the changes have been introduced later. The reason I tried to shorten then chunk length (specified in hexadecimal) was to meet the byte array size 16 as per the [condition](https://github.com/golang/go/blob/master/src/net/http/internal/chunked.go#L255). Though that leads to EOF errors.

@AlexVulaj and I had a debug session for the same, I just pushed the code so that it's available to test locally.

> I think we should just comment out that particular test data with a comment that references the above about it not existing and linking to the code etc.
Are we inclining to just comment it out & specify the reason for that?

Reply:
@AlexVulaj I have been considering your suggestion of keeping this test case but changing the expected response from "Ok" to the error, while looking deeper I realized that [chunked_test.go](https://github.com/golang/go/blob/master/src/net/http/internal/chunked_test.go#L147-L155) already has one such test case, so, I now feel it might not be necessarily required for gorilla/http. Let me know your thoughts :)

Reply:
I'm in favor of @coreydaley 's suggestion here. Let's go with that for now and we can always revisit this if necessary.